<?xml version="1.0"?>
<launch>
  <!-- By default we do not overwrite the URDF. Change the following to true to change the default behavior -->
  <!-- The name of the parameter under which the URDF is loaded -->
  <arg name="robot_description" default="robot_description"/>
  <arg name="trajopt_description" default="trajopt_description"/>
  <arg name="plotting" default="true"/>
  <arg name="rviz" default="true"/>
  <arg name="testing" default="false"/>
  <arg name="steps" default="12"/>
  <arg name="box_size" default="0.01"/>
  <arg name="update_start_state" default="false"/>
  <arg name="use_continuous" default="false"/>

  <!-- Load universal robot description format (URDF) -->
  <param name="$(arg robot_description)" command="$(find xacro)/xacro --inorder '$(find tesseract_ros_examples)/urdf/online_planning_example.xacro'" />

  <!-- The semantic description that corresponds to the URDF -->
  <param name="$(arg robot_description)_semantic" textfile="$(find tesseract_ros_examples)/config/online_planning_example.srdf" />

  <group unless="$(arg testing)">
    <!-- If needed, broadcast static tf for robot root -->
    <node pkg="tf" type="static_transform_publisher" name="virtual_joint_broadcaster_0" args="0 0 0 0 0 0 world base_link 100" />

    <node pkg="tesseract_ros_examples" type="tesseract_ros_examples_online_planning_example_node" name="online_planning_example_node" output="screen">
      <param name="plotting" type="bool" value="$(arg plotting)"/>
      <param name="rviz" type="bool" value="$(arg rviz)"/>
      <param name="steps" type="int" value="$(arg steps)"/>
      <param name="box_size" type="double" value="$(arg box_size)"/>
      <param name="update_start_state" type="bool" value="$(arg update_start_state)"/>
      <param name="use_continuous" type="bool" value="$(arg use_continuous)"/>
    </node>

    <!-- Launch visualization -->
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="online_planning_example_rviz"
         args="-d $(find tesseract_ros_examples)/config/online_example.rviz" />

    <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" respawn="true"/>
  </group>
</launch>
